### PR TITLE
RSDK-875 migrate rdk proto methods

### PR DESCRIPTION
**Pull request by maximpertsov**
JIRA: https://viam.atlassian.net/browse/RSDK-875

Initial release of typescript SDK - packages all web proto methods used by RDK: https://github.com/viamrobotics/rdk/pull/1601

TODO:
* [x] set test workflow branch to `@main` before merging

### CODE REVIEW

**Comment by micheal-parks** [comment]
Could we add separate ESLint and TS jobs in CI like we do for the rest of our repos? A good example of this would be the `pullrequest.yml` workflow within the PRIME repo.

**Comment by edaniels** [comment]
I'd like to stop doing that. It increases concurrency and adds unnecessary load. I removed it in rdk and those checks are fast anyway in an SDK like this so let's keep it the same

**Review by edaniels — APPROVED** [review]
LGTM

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
+module.exports = {
```
Comment (maximpertsov) [inline .eslintrc.js:1]: @micheal-parks I copied this as-is from three-inspect - let me know if any there are any settings that don't apply here.

**Review by micheal-parks — APPROVED** [review]

File: .github/workflows/pullrequest.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+name: Pull Request Update
+
+on:
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  test:
+    # TODO: lock to @main branch before merging
```
Comment (maximpertsov) [inline .github/workflows/pullrequest.yml:10]:
```suggestion
```

**Pull request merged**